### PR TITLE
Move of logs out of apiserver and into component-base

### PR DESCRIPTION
**Pull request by Klaven**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

/kind cleanup

**What this PR does / why we need it**:
This is a continuation of the component-base started in this https://github.com/kubernetes/kubernetes/pull/72569 PR. This PR moves the `k8s.io/apiserver/pkg/util/logs` pkg to `k8s.io/component-base/cli/logs` as detailed in the component-base backlog as outlined by https://github.com/kubernetes/community/pull/3008.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
this is a PR directly regarding the above working group.
**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by luxas — APPROVED** [review]
/approve
Thanks @Klaven! This looks good to me on a high level :+1:

I think we need to update a few import restriction files etc. so I'll wait with the LGTM until all tests are passing.

/assign @dims 
who have been working on this stuff as well.

/assign @thockin 
for approval. With this effort we hope to fixup the TODO from PR https://github.com/kubernetes/kubernetes/pull/240 soon-ish :wink:

**Comment by Klaven** [comment]
I will fix the verify imports. Let me know if you want me to change the directory structure and I will get it updated.

**Comment by dims** [comment]
+1 to `component-base/logging` @luxas (Naming is hard!!!)

**Comment by Klaven** [comment]
consider it done. as soon as I get the time I will have this updated and pushed.

**Comment by Klaven** [comment]
per a conversation we had on slack. we went with `/component-base/logs` due to naming.

File: cmd/kube-controller-manager/controller-manager.go [hunk 1]
```diff
@@ -26,8 +26,8 @@ import (
 	"os"
 	"time"
 
-	"k8s.io/apiserver/pkg/util/logs"
 	"k8s.io/kubernetes/cmd/kube-controller-manager/app"
+	"k8s.io/kubernetes/pkg/kubectl/util/logs"
```
Comment (luxas) [inline cmd/kube-controller-manager/controller-manager.go]: ??

File: hack/import-restrictions.yaml [hunk 2]
```diff
@@ -94,6 +94,7 @@
   - k8s.io/kube-openapi
   - k8s.io/utils
   - k8s.io/klog
+  - k8s.io/component-base
```
Comment (luxas) [inline hack/import-restrictions.yaml]: nit: add this in alphabetical order (I realize klog violate this, but anyway...). For all of these.

**Review by luxas — COMMENTED** [review]
Thanks for the update!
With regards to commit structure for refactors, please follow this pattern:
 - code changes (in one or many commits, here one only is necessary)
 - policy file updates (in this case; import restrictions)
 - autogenerated files (bazel and godeps)

That makes it way easier for the reviewer to wrap their head around the PR once they grow even larger  than this.

File: cmd/kube-controller-manager/controller-manager.go (hunk 1 above)
Comment (Klaven) [reply cmd/kube-controller-manager/controller-manager.go]: yeah... I have no idea. I will fix this.

**Comment by Klaven** [comment]
> Thanks for the update!
> With regards to commit structure for refactors, please follow this pattern:
> 
>     * code changes (in one or many commits, here one only is necessary)
> 
>     * policy file updates (in this case; import restrictions)
> 
>     * autogenerated files (bazel and godeps)
> 
> 
> That makes it way easier for the reviewer to wrap their head around the PR once they grow even larger than this.

@luxas do you want me to make a new PR with it done like this? or is this for future commits like this?

**Comment by luxas** [comment]
> @luxas do you want me to make a new PR with it done like this? or is this for future commits like this?

Do this in this PR please. Thanks!

**Review by luxas — APPROVED** [review]
/lgtm

**Pull request merged**